usrsctplib/netinet/sctp_usrreq.c

```diff
@@ -8179,11 +8179,13 @@ sctpconn_connect(struct socket *so, struct sockaddr *addr)
 		/* Gak! no memory */
 		goto out_now;
 	}
-	if (stcb->sctp_ep->sctp_flags & SCTP_PCB_FLAGS_TCPTYPE) {
+	SCTP_INP_RLOCK(stcb->sctp_ep);
```

Comment:
You can't aquire an `inp` lock here since you own already the `stcb` lock. This would result in a LOR.